crates/dbt-common/src/io_args.rs

```diff
@@ -606,41 +606,45 @@ pub fn check_var(vars: &str) -> Result<BTreeMap<String, Value>, String> {
     // Strip outer quotes if present
     let vars = vars.trim().trim_matches('\'');
 
-    // Check if the input is already wrapped in curly braces
-    let yaml_str = if vars.trim().starts_with('{') {
-        vars.to_string()
-    } else {
-        // Handle single key-value pair separated by a colon
-        if vars.trim().matches(':').count() != 1 {
-            return Err(format!(
-                "Invalid key-value pair: '{vars}'. Expected format: 'key: value'."
-            ));
-        }
-        vars.to_string()
-    };
-
     // Try parsing as YAML first
-    match dbt_serde_yaml::from_str::<BTreeMap<String, Value>>(&yaml_str) {
+    match dbt_serde_yaml::from_str::<BTreeMap<String, Value>>(vars) {
         Ok(btree) => {
-            // Disallow the '{key:value}' format for flow-style YAML syntax
-            // to prevent key:value: None interpretation: https://stackoverflow.com/a/70909331
-            for key in btree.keys() {
-                if key.contains(':') {
-                    return Err(format!(
-                        "Invalid key-value pair: '{key}'. Value must start with a space after colon."
-                    ));
+            // Check for flow-style YAML without space after colon, e.g., '{key:value}'
+            // This is an invalid YAML syntax according to the spec but might be leniently parsed by some libraries.
+            // We want to explicitly disallow it to prevent incorrect interpretations (like 'value' being null).
+            if vars.starts_with('{') && vars.contains(':') {
+                let potential_pairs: Vec<&str> = vars
+                    .trim_matches('{')
+                    .trim_matches('}')
+                    .split(',')
+                    .collect();
+                for pair in potential_pairs {
+                    let trimmed_pair = pair.trim(); // Trim leading/trailing whitespace from each pair
+                    if let Some(colon_idx) = trimmed_pair.find(':') {
+                        // Check if the character immediately after the colon is not a space
+                        // and ensure it's not the end of the string (e.g., "key:")
+                        if colon_idx + 1 < trimmed_pair.len()
+                            && trimmed_pair.chars().nth(colon_idx + 1) != Some(' ')
+                        {
+                            return Err("Invalid YAML format: Missing space after colon in flow-style mapping. Example: '{key: value}'".to_string());
+                        }
+                    }
                 }
             }
+            // Disallow the '{key:value}' format for flow-style YAML syntax
+            // to prevent key:value: None interpretation: https://stackoverflow.com/a/70909331
```

Comment:
This comment is misplaced here but could be removed if you move the stackoverflow link to the comment before `vars.starts_with(...`

Reply:
I moved the comment before `vars.tarts_with(...`.